packages/go_router/lib/src/configuration.dart

```diff
@@ -80,6 +80,14 @@ class RouteConfiguration {
                 ...allowedKeys..add(route.navigatorKey)
               ],
             );
+          } else if (route is StatefulShellRoute) {
+            checkParentNavigatorKeys(
+              route.routes,
+              <GlobalKey<NavigatorState>>[
+                ...allowedKeys,
+                ...route.navigatorKeys,
```

Comment:
This does not look right, the allowed key would depend on which subroute is active.

Reply:
No, it doesn't, good catch! I've changed this now.